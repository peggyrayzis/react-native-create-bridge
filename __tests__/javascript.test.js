import path from 'path';
const fileOperations = require('../src/file-operations');
const { readFile, parseFile } = fileOperations;

describe('JS: Combined', () => {
  const templateName = 'TestModule';
  const readDirPath = path.join(__dirname, '..', 'templates', 'combined', 'js');

  it('creates a TemplateNativeModule.js', async () => {
    const fileData = await readFile('TemplateNativeModule.js', readDirPath);
    const parsedFile = parseFile(fileData, { templateName });
    expect(parsedFile).toMatchSnapshot();
  });

  it('creates a TemplateNativeView.js', async () => {
    const fileData = await readFile('TemplateNativeView.js', readDirPath);
    const parsedFile = parseFile(fileData, { templateName });
    expect(parsedFile).toMatchSnapshot();
  });
});

describe('JS: Native Modules', () => {
  const templateName = 'TestModule';
  const readDirPath = path.join(__dirname, '..', 'templates', 'modules', 'js');

  it('creates a TemplateNativeModule.js', async () => {
    const fileData = await readFile('TemplateNativeModule.js', readDirPath);
    const parsedFile = parseFile(fileData, { templateName });
    expect(parsedFile).toMatchSnapshot();
  });
});

describe('JS: UI Components', () => {
  const templateName = 'TestModule';
  const readDirPath = path.join(
    __dirname,
    '..',
    'templates',
    'ui-components',
    'js',
  );

  it('creates a TemplateNativeView.js', async () => {
    const fileData = await readFile('TemplateNativeView.js', readDirPath);
    const parsedFile = parseFile(fileData, { templateName });
    expect(parsedFile).toMatchSnapshot();
  });
});

describe('JS: Semver', () => {
  const config = {
    templateName: 'SemverModule',
    rnVersion: "^0.49.1"
  }
  const readDirPath = path.join(
    __dirname,
    '..',
    'templates',
    'ui-components',
    'js',
  );

  it('correctly parses React Native version', async () => {
    const fileData = await readFile('TemplateNativeView.js', readDirPath);
    const parsedFile = parseFile(fileData, config);
    expect(parsedFile).toMatchSnapshot();
  });
});
