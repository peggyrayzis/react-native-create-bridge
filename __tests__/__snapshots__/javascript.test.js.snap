// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS: Combined creates a TemplateNativeModule.js 1`] = `
"//  Created by react-native-create-bridge

import { NativeModules } from 'react-native'

const { TestModule } = NativeModules

export default {
  exampleMethod () {
    return TestModule.exampleMethod()
  },

  EXAMPLE_CONSTANT: TestModule.EXAMPLE_CONSTANT
}
"
`;

exports[`JS: Combined creates a TemplateNativeView.js 1`] = `
"//  Created by react-native-create-bridge

import React, { Component } from 'react'
import { requireNativeComponent } from 'react-native'

const TestModule = requireNativeComponent('TestModule', TestModuleView)

export default class TestModuleView extends Component {
  render () {
    return <TestModule {...this.props} />
  }
}

TestModuleView.propTypes = {
  exampleProp: React.PropTypes.any
}
"
`;

exports[`JS: Native Modules creates a TemplateNativeModule.js 1`] = `
"//  Created by react-native-create-bridge

import { NativeModules } from 'react-native'

const { TestModule } = NativeModules

export default {
  exampleMethod () {
    return TestModule.exampleMethod()
  },

  EXAMPLE_CONSTANT: TestModule.EXAMPLE_CONSTANT
}
"
`;

exports[`JS: Semver correctly parses React Native version 1`] = `
"//  Created by react-native-create-bridge

import React, { Component } from 'react'
import { requireNativeComponent } from 'react-native'

const SemverModule = requireNativeComponent('SemverModule', SemverModuleView)

export default class SemverModuleView extends Component {
  render () {
    return <SemverModule {...this.props} />
  }
}

SemverModuleView.propTypes = {
  exampleProp: React.PropTypes.string
}
"
`;

exports[`JS: UI Components creates a TemplateNativeView.js 1`] = `
"//  Created by react-native-create-bridge

import React, { Component } from 'react'
import { requireNativeComponent } from 'react-native'

const TestModule = requireNativeComponent('TestModule', TestModuleView)

export default class TestModuleView extends Component {
  render () {
    return <TestModule {...this.props} />
  }
}

TestModuleView.propTypes = {
  exampleProp: React.PropTypes.string
}
"
`;
